client/src/ts/dd_main.ts

```diff
@@ -160,7 +160,9 @@ function updateDate(): void {
 // Initializations
 
 let prevDir = 4;
-const ws = new WebSocket('ws://localhost:8080/ws');
+const ws = new WebSocket(
+  'ws://' + window.location.hostname + ':' + window.location.port + '/ws'
+);
```

Comment:
Can the changes in this file be done in a separate PR? It seems independent from all the main changes this PR is achieving.

Reply:
This PR depends on this change since otherwise the client webpage tries to connect to the incorrect address. Also since it's such a small and uncontroversial change we can just get it all in at once and avoid the extra overhead

Reply:
If the PR depends on it, then this change can go in before this current PR. I agree it's not a big deal but I just wanted to mention that. Overall, awesome job man! :D